mmdet/models/anchor_heads/guided_anchor_head.py

```diff
@@ -0,0 +1,542 @@
+from __future__ import division
+
+import numpy as np
+import torch
+import torch.nn as nn
+from mmcv.cnn import normal_init
+
+from mmdet.core import (AnchorGenerator, anchor_target, ga_loc_target,
+                        ga_shape_target, delta2bbox, multi_apply,
+                        weighted_smoothl1, weighted_sigmoid_focal_loss,
+                        weighted_cross_entropy, weighted_binary_cross_entropy,
+                        weighted_bounded_iou_loss, multiclass_nms)
+from mmdet.ops import DeformConv, MaskedConv2d
+from ..registry import HEADS
+from ..utils import bias_init_with_prob
+
+
+class FeatureAdaption(nn.Module):
+
+    def __init__(self,
+                 in_channels,
+                 out_channels,
+                 kernel_size=3,
+                 adaption_groups=4):
+        super(FeatureAdaption, self).__init__()
+        offset_channels = kernel_size * kernel_size * 2
+        self.conv_offset = nn.Conv2d(
+            2, adaption_groups * offset_channels, 1, bias=False)
+        self.conv_adaption = DeformConv(
+            in_channels,
+            out_channels,
+            kernel_size=kernel_size,
+            padding=int((kernel_size - 1) / 2),
+            deformable_groups=adaption_groups)
+        self.relu = nn.ReLU(inplace=True)
+
+    def init_weights(self):
+        normal_init(self.conv_offset, std=0.1)
+        normal_init(self.conv_adaption, std=0.01)
+
+    def forward(self, x, shape):
+        offset = self.conv_offset(shape.detach())
+        x = self.relu(self.conv_adaption(x, offset))
+        return x
+
+
+@HEADS.register_module
+class GuidedAnchorHead(nn.Module):
+    """Guided-Anchor-based head (GA-RPN, GA-RetinaNet, etc.).
+
+    Args:
+        in_channels (int): Number of channels in the input feature map.
+        feat_channels (int): Number of channels of the feature map.
+        octave_base_scale: base octave scale of each level of feature map.
```

Comment:
The format is `arg name (type): description` and the first character of description should be in upper case.